.github/workflows/ci.yaml

```diff
@@ -191,13 +195,18 @@ jobs:
         run: nox -s safety
 
   build:
-    if: github.event_name == 'push' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/'))
-    needs: [tests, linting, formatting, safety]
+    needs: [ tests, linting, formatting, safety]
     runs-on: ubuntu-22.04
+    env:
+      PUSH_DOCKER_IMAGE_TO_HARBOR: ${{ (github.ref == 'refs/heads/main') || (inputs.push-docker-image-to-harbor) && 'true' || 'false' }}
```

Comment:
```suggestion
      PUSH_DOCKER_IMAGE_TO_HARBOR: ${{ (github.ref == 'refs/heads/main') || startsWith(github.ref, 'refs/tags/') || (inputs.push-docker-image-to-harbor) && 'true' || 'false' }}
```

I didn't mention it explicitly on Slack but we should also keep the condition for pushing to Harbor when a new tag is created. In practice, we will do this when we release a new version of the API. Pushes to main correspond to the latest "development" version of the API, and the manual workflow submission is a useful debugging tool.